mongosql/src/ast/rewrites/extended_unwind_rewrite.rs

```diff
@@ -0,0 +1,245 @@
+                        }
+                    }
+                }
+                let global_outer = global_outer.unwrap_or(false);
+                if paths.is_none() {
+                    self.error = Some(Error::UnwindSourceWithoutPath);
+                    return ast::Datasource::ExtendedUnwind(ExtendedUnwindSource {
+                        datasource: source,
+                        options: Vec::new(),
+                    });
+                }
+                let mut paths = paths.unwrap();
+                // if there is only one path and there are no local unwind options, this is an old
+                // style UnwindSource that we can immediately convert to a normal UnwindSource.
+                // In particular, this keeps us from prefixing the INDEX field with the path.
+                if paths.len() == 1 && paths[0].iter().all(|p| p.options.is_empty()) {
+                    create_simple_unwind_datasource(
+                        source,
+                        paths.remove(0),
+                        global_index,
+                        global_outer,
+                    )
+                } else {
+                    create_unwind_datasource(*source, paths, global_index, global_outer)
+                }
+            }
+            _ => data_source.walk(self),
+        }
+    }
+}
+
+// This converts a vector of strings representing a path into a nested SubpathExpr.
+// Each element can be thought of as separated by `.` in the path. By representing it as a
+// Vec, we can not worry about parts that contain `.` in them.
+fn path_vec_to_path(mut path: Vec<String>) -> Expression {
+    if path.len() == 1 {
+        return Expression::Identifier(path.remove(0));
+    }
+    let mut ret = Expression::Identifier(path.remove(0));
+    for p in path.into_iter() {
+        ret = Expression::Subpath(SubpathExpr {
+            expr: Box::new(ret),
+            subpath: p.to_string(),
+        });
+    }
+    ret
+}
+
+// Get options takes the ParthPartOptions and the path and returns a vector of UnwindOptions for
+// normal UnwindSources.
+fn get_options(
+    options: Vec<UnwindPathPartOption>,
+    path: Vec<String>,
+    index_prefix: &str,
+    global_index: Option<&String>,
+    global_outer: bool,
+) -> Vec<UnwindOption> {
+    let mut ret = Vec::new();
+    let mut found_index = false;
+    let mut found_outer = false;
+    for option in options.into_iter() {
+        match option {
+            UnwindPathPartOption::Index(i) => {
+                ret.push(UnwindOption::Index(i));
+                found_index = true;
+            }
+            UnwindPathPartOption::Outer(o) => {
+                ret.push(UnwindOption::Outer(o));
+                found_outer = true;
+            }
+        }
+    }
+
+    if !found_index && global_index.is_some() {
+        ret.push(UnwindOption::Index(format!(
+            "{}_{}",
+            index_prefix,
+            global_index.unwrap()
+        )));
+    }
```

Comment:
```suggestion
    if !found_index {
        if let Some(gi) = &global_index {
            ret.push(UnwindOption::Index(format!("{}_{}", index_prefix, gi)));
        }
    }
```
nit: this could help avoid having the `unwrap()` call

Reply:
Surprised clippy didn't complain